plugins/device/rest.go

```diff
@@ -67,9 +79,68 @@ func deviceAddHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	peerInfo, err = peer.GetPeer(peerID)
 	if err != nil {
-		logger.WithError(err).Error("Failed to get peer from store")
+		logger.WithError(err).WithField("peerid", peerID).Error("Failed to get peer from store")
 		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, "Failed to get peer from store")
 		return
 	}
 	restutils.SendHTTPResponse(ctx, w, http.StatusOK, peerInfo)
 }
+
+func peerEditZoneHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
You can implement editing zone as part of peer edit handler as Zone is property or characteristic of a peer first and then devices. That's also much less code, which theoretically translates to less bugs and less places to fix things later.